pkg/testers/exec/exec.go

```diff
@@ -64,8 +65,23 @@ func (t *Tester) Execute() error {
 	return t.Test()
 }
 
+func expandEnv(args []string) []string {
+	expandedArgs := make([]string, len(args))
+	for i, arg := range args {
+		// best effort handle literal dollar for backward compatibility
+		// this is not an all-purpose shell special character handler
+		if strings.Contains(arg, `\$`) {
+			expandedArgs[i] = strings.ReplaceAll(arg, `\$`, `$`)
+		} else {
+			expandedArgs[i] = os.ExpandEnv(arg)
+		}
```

Comment:
```suggestion
		if strings.Contains(arg, `\$`) {
			arg = strings.ReplaceAll(arg, `\$`, `$`)
		}
		expandedArgs[i] = os.ExpandEnv(arg)
```
flows a little better imo

Reply:
we explicitly don't want to expand escaped `$` hence the `else`?